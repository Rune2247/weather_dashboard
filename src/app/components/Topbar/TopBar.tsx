"use client"
import React, { useEffect, useState } from "react"
import { styles } from "./TopBar.style"

const WeatherIcon = () => (
	<span style={styles.icon}>
		<span
			role="img"
			aria-label="weather"
		>
			☀️
		</span>
	</span>
)

export const TopBar: React.FC = () => {
	const [now, setNow] = useState(new Date())

	useEffect(() => {
		const interval = setInterval(() => setNow(new Date()), 1000)
		return () => clearInterval(interval)
	}, [])

	const pad = (n: number) => n.toString().padStart(2, "0")
	const time = `${pad(now.getHours())}:${pad(now.getMinutes())}`
	const date = `${pad(now.getDate())}-${pad(
		now.getMonth() + 1
	)}-${now.getFullYear()}`

	return (
		<header style={{ ...styles.header, position: "fixed", top: 0, left: 0 }}>
			<WeatherIcon />
			<span style={styles.title}>Weather Dashboard</span>
			<div style={{ ...styles.timeBox, textAlign: "right" }}>
				<div style={styles.time}>{time}</div>
				<div style={styles.date}>{date}</div>
			</div>
		</header>
	)
}
