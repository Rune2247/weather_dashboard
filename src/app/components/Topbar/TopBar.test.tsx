import { render } from "@testing-library/react"
import { TopBar } from "./TopBar"

describe("TopBar", () => {
	it("should match snapshot", () => {
		const { baseElement } = render(<TopBar />)
		expect(baseElement).toMatchSnapshot()
	})

	it("renders the title", () => {
		const { getByText } = render(<TopBar />)
		expect(getByText("Weather Dashboard")).toBeInTheDocument()
	})
})
