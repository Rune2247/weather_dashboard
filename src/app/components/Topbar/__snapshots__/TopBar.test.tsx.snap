// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar should match snapshot 1`] = `
<body>
  <div>
    <header
      style="width: 100vw; background: rgb(255, 255, 255); color: rgb(34, 34, 34); box-shadow: 0 2px 8px rgba(0,0,0,0.04); padding: 0.75rem 2rem; display: flex; align-items: center; position: fixed; top: 0px; left: 0px; z-index: 100; min-height: 56px; font-weight: 600; font-size: 1.35rem; letter-spacing: 0.5px; border-bottom: 1px solid #eee;"
    >
      <span
        style="font-size: 28px; margin-right: 16px; display: flex; align-items: center;"
      >
        <span
          aria-label="weather"
          role="img"
        >
          ☀️
        </span>
      </span>
      <span
        style="flex: 1;"
      >
        Weather Dashboard
      </span>
      <div
        style="text-align: right; min-width: 80px; line-height: 1.1;"
      >
        <div
          style="font-size: 18px; font-weight: 600;"
        >
          18:15
        </div>
        <div
          style="font-size: 13px; color: rgb(136, 136, 136);"
        >
          29-05-2025
        </div>
      </div>
    </header>
  </div>
</body>
`;
